headless_zip.js

```diff
@@ -13,98 +13,112 @@
+    date.getFullYear().toString().substr(2, 3) +
+    '_' +
+    pad(date.getHours().toString()) +
+    pad(date.getMinutes().toString()) +
+    pad(date.getSeconds().toString()) +
+    '_' +
+    process.pid;
+  return timestamp;
 }
 
 function pad(numberString) {
-	// pads a single digit number with a leading 0
-	if(numberString.length == 1) {
-		return '0' + numberString
-	}
-	return numberString
+  // pads a single digit number with a leading 0
+  if (numberString.length == 1) {
+    return '0' + numberString;
+  }
+  return numberString;
 }
 
 module.exports.headlessZip = function headlessZip(dirToZip) {
-	var outputFileName;
-	if(dirToWriteTo) {
-		outputFileName = path.join(dirToWriteTo, 'nodeappmetrics' + timestamp() + '.hcd')
-	} else {
-		outputFileName = 'nodeappmetrics' + timestamp() + '.hcd'
-	}
+  var outputFileName;
+  if (dirToWriteTo) {
+    outputFileName = path.join(
+      dirToWriteTo,
+      'nodeappmetrics' + timestamp() + '.hcd'
+    );
+  } else {
+    outputFileName = 'nodeappmetrics' + timestamp() + '.hcd';
+  }
 
-	filesWritten.push(outputFileName)
-	if(filesWritten.length > filesToKeep) {
-		var earliest = filesWritten.shift()
-		deleteFile(earliest)
-	}
+  filesWritten.push(outputFileName);
+  if (filesWritten.length > filesToKeep) {
+    var earliest = filesWritten.shift();
+    deleteFile(earliest);
+  }
 
-	if(fs.existsSync(dirToZip)) {
-		var zip = new JSZip();
-		var files = fs.readdirSync(dirToZip)
-		for (var i = 0, len = files.length; i < len; i++) {
-			zip.file(files[i], fs.readFileSync(path.join(dirToZip, files[i])), {compression : "DEFLATE"})
-		}
-		fs.writeFileSync(outputFileName, zip.generate({type:"nodebuffer", compression:'DEFLATE'}));
-		deleteDir(dirToZip)
-	}
-}
+  if (fs.existsSync(dirToZip)) {
+    var zip = new JSZip();
+    var files = fs.readdirSync(dirToZip);
+    for (var i = 0, len = files.length; i < len; i++) {
+      zip.file(files[i], fs.readFileSync(path.join(dirToZip, files[i])), {
+        compression: 'DEFLATE',
+      });
+    }
+    fs.writeFileSync(
+      outputFileName,
+      zip.generate({ type: 'nodebuffer', compression: 'DEFLATE' })
+    );
+    deleteDir(dirToZip);
+  }
+};
 
 module.exports.tryZipOnExit = function tryZipOnExit() {
-	var outputDir = dirToWriteTo;
-	if(!outputDir) {
-		ouputDir = process.cwd().toString()
-	}
-	if(fs.existsSync(outputDir)) {
-		var files = fs.readdirSync(outputDir);
-		// Search for temporary output directory using pattern matching
-		for (var i = 0, len = files.length; i < len; i++) {
-			if(/tmp_(\w+)/.test(files[i].toString())) {
-				var dirToZip = path.join(outputDir, files[i])
-				this.headlessZip(dirToZip)
-				return;
-			}
-		}
-	}
-}
-
+  var outputDir = dirToWriteTo;
+  if (!outputDir) {
+    // FIXME(sam) not sure what to do here, this code probably never did
+    // anything but I don't want to enable it in error (btw,  cwd() always
+    // returns a string).
+    ouputDir = process.cwd().toString();
```

Comment:
See comment above. The code could be rewritten as `var outputDir = dirToWriteTo || process.cwd();`, but since code has never actually done anything, perhaps it should just be deleted?